packages/evm/contracts/BoostCore.sol

```diff
@@ -204,6 +227,46 @@ contract BoostCore is Ownable, ReentrancyGuard {
         return _boosts.length;
     }
 
+    /// @notice Get the incentives for a Boost
+    /// @param key The key composed of the Boost ID and the Incentive ID - keccak256(abi.encodePacked(boostId, incentiveId))
+    function getIncentive(bytes32 key) external view returns (IncentiveDisbursalInfo memory) {
+        return incentives[key];
+    }
+
+    /// @notice Returns the protocol fee and any remaining incentive value to the owner or budget
+    /// @param boostId The ID of the Boost
+    function clawback(bytes calldata data_, uint256 boostId, uint256 incentiveId) external nonReentrant {
+        // We want to check that the sender is authorized to clawback the incentive
```

Comment:
Be sure to do this auth check on the budget, not the incentive itself, and make sure the caller is only the admin instead of the manager if the `claimTarget` is unconstrained